src/tasks/platforms/common-platform-tasks.ts

```diff
@@ -72,4 +78,47 @@ export namespace CommonPlatformTasks {
     })
   }
 
+  export function oAuthProvidersExists(flags: any): Listr.ListrTask {
+    let kube = new KubeHelper(flags)
+    return {
+      title: 'Verify the existence of identity providers.',
+      enabled: () => isOpenshiftPlatformFamily(flags.platform) && isOAuthEnabled(flags),
+      task: async (ctx: any, task: any) => {
+        const providers = await kube.getOpenshiftAuthProviders()
+        if (!providers || providers.length === 0) {
+          ctx.highlightedMessages.push(`❗ ${ansi.yellow('[WARNING]')} 'os-oauth' flag was disabled, because Openshift oauth hasn't got any identity providers.`)
+          flags['os-oauth'] = false
+        }
+        task.title = `${task.title}...done.`
+      }
+    }
+  }
+
+  /**
+   * Checks if Openshift oAuth is disabled via operator custom resource.
+   * Returns true if Openshift oAuth is enabled (or omitted) and false if it is explicitly disabled.
+   */
+  async function isOAuthEnabled(flags: any): Promise<boolean> {
+    if (flags['che-operator-cr-yaml']) {
+      const cheOperatorCrYamlPath = flags['che-operator-cr-yaml']
+      if (fs.existsSync(cheOperatorCrYamlPath)) {
+        const cr = yaml.safeLoad(fs.readFileSync(cheOperatorCrYamlPath).toString())
+        if (cr && cr.spec && cr.spec.auth && cr.spec.auth.openShiftoAuth) {
```

Comment:
I am doubt if `&& cr.spec.auth.openShiftoAuth` is correct check.
What if user defined `cr.spec.auth.openShiftoAuth: false` ?